src/components/Router/StepComponentMap.ts

```diff
@@ -120,39 +124,46 @@ const buildCaptureStepComponents = (
   const documentStep = findStep('document')
 
   const complete = mobileFlow ? [ClientSuccess] : [Complete]
+  const captureStepTypes = new Set(['document', 'poa', 'face'])
+  const firstCaptureStepType = steps.filter((step) =>
+    captureStepTypes.has(step?.type)
+  )[0]?.type
+  const showCrossDeviceClientIntroForFaceStep =
+    mobileFlow && firstCaptureStepType === 'face'
+  const showCrossDeviceClientIntroForDocumentStep =
+    mobileFlow && firstCaptureStepType === 'document'
+  const showCrossDeviceClientIntroForPoaStep =
+    mobileFlow && firstCaptureStepType === 'poa'
 
   return {
     welcome: [Welcome],
     userConsent: [UserConsent],
-    face: buildFaceComponents(faceStep, deviceHasCameraSupport, mobileFlow),
+    face: buildFaceComponents(
```

Comment:
Maybe something like this would simplify the logic in this component?
```
face: [...crossDeviceClientIntro, ...buildFaceComponents(faceStep, deviceHasCameraSupport, mobileFlow)],
```

And crossDeviceClientIntro would be
```
const crossDeviceClientIntro = mobileFlow ? [CrossDeviceIntro] : []

Reply:
It looks like it's not really required after my refactor, but will definitely keep this in mind. :)